### PR TITLE
Add MCP4728 output component

### PR DESCRIPTION
**Pull request by brotherdust**
Signed-off-by: Jarad Olson <brotherdust@gmail.com>

## Description:

Add docs for mcp4728 output component

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#2030

## Checklist:

  - [X ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [X] Link added in `/index.rst` when creating new documents for new components or cookbook.

Question about the "link added": the link seems to be automatically generated. Is this acceptable?

### CODE REVIEW

**Comment by glmnet** [comment]
It's missing a link from index

File: components/output/mcp4728.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+------------------------
+
+- **address** (*Optional*, int): Manually specify the I2C address of
+  the DAC. Defaults to ``0x60``.
+- All other options from :ref:`Output <config-output>`.
+
+Usage with voltages higher than 3.3v
+------------------------------------
+
+In order to drive analog modules with voltages higher than 3.3v, use a `TTL bi-directionnal level
```
Comment (oxan) [inline components/output/mcp4728.rst]:
```suggestion
In order to drive analog modules with voltages higher than 3.3v, use a `TTL bi-directional level
```

**Review by oxan — COMMENTED** [review]
Noticed a typo :)

File: components/output/mcp4728.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      id: dac_D
+      channel: D
+
+
+Configuration variables:
+------------------------
+
+- **address** (*Optional*, int): Manually specify the I2C address of
+  the DAC. Defaults to ``0x60``.
+- All other options from :ref:`Output <config-output>`.
```
Comment (jesserockz) [inline components/output/mcp4728.rst:48]: This needs 2 separate `Configuration variables` blocks:

```suggestion
Component/Hub
-------------

Configuration variables:
************************

- **address** (*Optional*, int): Manually specify the I2C address of
  the DAC. Defaults to ``0x60``.


Output
------

Configuration variables:
************************

- **channel** (**Required**, string): The channel of this output. Must be one of ``A``, ``B``, ``C`` or ``D``.
- All other options from :ref:`Output <config-output>`.
```

**Pull request closed without merging**